src/outputs/harbor_output.ml

```diff
@@ -535,11 +523,8 @@ module Make (T : T) = struct
               true )
             else false
           in
-          burst_data <- (
-            let bd = Strings.append burst_data b in
-            let len = Strings.length bd in
-            Strings.sub bd (max 0 (len-burst)) (min len burst)
-          );
+          Strings.Mutable.append_strings burst_data b;
+          Strings.Mutable.keep burst_data burst;
```

Comment:
My `sub` is more simple than `keep` here: since we now have string views, I would simply remove `keep` and use `sub`.

Reply:
They are very similar indeed. I added `keep` to simplify the semantics of the code, making it easier to read without having to do the math (and potentially get it wrong).